qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -108,8 +109,8 @@ class OptimizationResult:
 
     Status = OptimizationResultStatus
 
-    def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
-                 results: Optional[Any] = None,
+    def __init__(self, x: Union[List[float], np.ndarray], fval: float,
+                 results: Any,
```

Comment:
I guess we can keep `results` as optional?

Reply:
It's possible to keep it as optional, but all algorithms pass something to `results` now. So, I removed `Optional`. Do you want keep it as optional?

Reply:
I would keep it optional but I'm with either solution.

Reply:
OK. Then, I will make it optional.